pkg/condition/conditions.go

```diff
@@ -106,6 +118,35 @@ func (c *condition) IsWindowsIPAMEnabled() bool {
 	return false
 }
 
+func (c *condition) IsWindowsPrefixDelegationEnabled() bool {
```

Comment:
Where will this function be called? I assume this should be called after the check on IsWindowsTurnedOn passes. If so, seems no reason to check if Windows is on again here.

Reply:
This will be called by resource providers to know how to initialize their pools and no call is made to `IsWindowsIPAMEnabled()` before this one. So adding here to make sure IPAM is indeed enabled.